### PR TITLE
fix(http body saver): properly deal with EOF terminated bodies

### PR DESCRIPTION
**Pull request by bassosimone**
See https://github.com/ooni/probe-engine/issues/1191

### CODE REVIEW

**Review by hellais — APPROVED** [review]
LGTM!

File: internal/engine/netx/httptransport/saver.go [hunk 1]
```diff
@@ -139,6 +141,22 @@ func (txp SaverBodyHTTPTransport) RoundTrip(req *http.Request) (*http.Response,
...
+// ignoreExpectedEOF converts an error signalling the end of the body
+// into a success. We know that we are in such condition when the
+// resp.Close hint flag is set to true. (Thanks, stdlib!)
+//
+// See https://github.com/ooni/probe-engine/issues/1191 for an analysis
+// of how this error was impacting measurements and data quality.
+func ignoreExpectedEOF(err error, resp *http.Response) error {
+	if err == nil {
+		return nil
+	}
```
Comment (hellais) [inline internal/engine/netx/httptransport/saver.go:153]: Isn't this redundant?

Comment (hellais) [reply internal/engine/netx/httptransport/saver.go:153]: If the err is `nil` you should be able to return it as `nil` nonetheless, no?

Comment (bassosimone) [reply internal/engine/netx/httptransport/saver.go:153]: yes, I prefer to have it explicit in the code

**Pull request merged**